File: Valid Palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        int len = s.length();
        s = s.toLowerCase();
        StringBuilder stringBuilder = new StringBuilder();
        for(int i=0; i<len; i++){
            int ascii = (int) s.charAt(i);
            if((ascii > 96 && ascii<123) || (ascii>47 && ascii<58)){
                stringBuilder.append(s.charAt(i));
            }
        }
        String converted = stringBuilder.toString();
        len = converted.length();
        int i=0; int j = len-1;
        while (i!=j && i<len){
            if(converted.charAt(i)==converted.charAt(j)){
                i++;
                j--;
            }else{
                return false;
            }
        }
        return true;
    }
}

File: Valid Palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = s.lower()
        new_s = ""
        for ch in s:
            if (ord(ch) >= 97 and ord(ch) <= 122) or (ord(ch) >= 48 and ord(ch) <= 57):
                new_s += ch
        i, j = 0, len(new_s)-1
        while i <= j:
            if new_s[i] != new_s[j]:
                return False
            i += 1
            j -= 1
        return True